app/code/Magento/Cron/Model/DeadlockRetrier.php

```diff
@@ -30,6 +44,7 @@ public function execute(callable $callback, AdapterInterface $connection)
             try {
                 return $callback();
             } catch (DeadlockException $e) {
+                $this->logger->warning(sprintf("Deadlock detected in cron cleanup: %s", $e->getMessage()));
```

Comment:
Deadlock retrier does not retry only DELETE queries but UPDATE queries as well.
https://github.com/magento/magento2/blob/6dd0bb69333c549d2267cfc7a054eb66da1deadf/app/code/Magento/Cron/Model/Schedule.php#L282-L291

```suggestion
                $this->logger->warning(sprintf("Deadlock detected in cron: %s", $e->getMessage()));
```